CMakeLists.txt

```diff
@@ -251,15 +251,18 @@ if(DEACTIVATE_AVX2)
 endif()
 
 # flags
-# @TODO: set -Wall
+# Set -Wall and other useful warning flags.
+if(CMAKE_C_COMPILER_ID STREQUAL GNU OR CMAKE_C_COMPILER_ID STREQUAL Clang OR CMAKE_C_COMPILER_ID STREQUAL Intel)
+    add_compile_options(-Wall -Wwrite-strings -Wno-unused-function)
+endif(CMAKE_C_COMPILER_ID STREQUAL GNU OR CMAKE_C_COMPILER_ID STREQUAL Clang OR CMAKE_C_COMPILER_ID STREQUAL Intel)
+
 # @NOTE: -O3 is enabled in Release mode (CMAKE_BUILD_TYPE="Release")
 
-# Set the "-msse2" build flag only if the CMAKE_C_FLAGS is not already set.
-# Probably "-msse2" should be appended to CMAKE_C_FLAGS_RELEASE.
+# Set the "-msse2" build flag if supported.
 if(CMAKE_C_COMPILER_ID STREQUAL GNU OR CMAKE_C_COMPILER_ID STREQUAL Clang OR CMAKE_C_COMPILER_ID STREQUAL Intel)
-     if(NOT CMAKE_C_FLAGS AND COMPILER_SUPPORT_SSE2)
-         set(CMAKE_C_FLAGS -msse2 CACHE STRING "C flags." FORCE)
-     endif(NOT CMAKE_C_FLAGS AND COMPILER_SUPPORT_SSE2)
+     if(COMPILER_SUPPORT_SSE2)
+         add_compile_options(-msse2)
+     endif(COMPILER_SUPPORT_SSE2)
```

Comment:
Does add_compile_options() require a cmake > 2.8.10?  This is the current minimum required.

Reply:
Good catch, seems it was added in cmake 2.8.12. Should we require cmake >= 2.8.12 or should I revert my changes to `CMakeLists.txt`?

Reply:
Fortunately it seems like [Ubuntu trusty has 2.8.12](https://packages.ubuntu.com/trusty/cmake), so let's make a bump.  Could you do that in main CMakeLists.txt?